lib/saml/user.rb

```diff
@@ -0,0 +1,63 @@
+# frozen_string_literal: true
+require 'saml/user_attributes/id_me'
+require 'saml/user_attributes/mhv'
+
+module SAML
+  class User
+    attr_reader :saml_response, :attributes, :authn_context, :decorated
+
+    def initialize(saml_response)
+      @saml_response = saml_response
+      @authn_context = saml_response&.settings&.authn_context
+      @attributes = saml_response.attributes
+      @decorated = decorator_constant.new(self)
+    end
+
+    def last_signed_in
+      Time.current.utc
+    end
+
+    def loa
+      { current: loa_current, highest: loa_highest(attributes) }
+    end
+
+    def mhv_user?
+      @authn_context == 'mhv'
+    end
+
+    def dslogon_user?
+      @authn_context == 'dslogon'
+    end
+
+    def idme_user?
+      !mhv_user? && !dslogon_user?
```

Comment:
This logic relies on the assumption that we only have 3 ways of logging in, which might not be true forever.

It could instead be written as:
```ruby
LOA::MAPPING.key?(@authn_context)
```

Not super, urgent and works either way.

Reply:
Yes, I'll be change all of this, just wrote it up like this for now to draw attention to it. None of those methods are even being used currently and I could probably rip them out to be honest.